### PR TITLE
DevWs for YookSmart containing containing ZigBee Window Shade Battery

### PR DESCRIPTION
**Pull request by JeremeLau**

### CODE REVIEW

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy [hunk 1]
```diff
@@ -27,8 +27,13 @@ metadata {
 
 		command "pause"
 
+        // IKEA
 		fingerprint manufacturer: "IKEA of Sweden", model: "KADRILJ roller blind", deviceJoinName: "IKEA Window Treatment" // raw description 01 0104 0202 00 09 0000 0001 0003 0004 0005 0020 0102 1000 FC7C 02 0019 1000 //IKEA KADRILJ Blinds
 		fingerprint manufacturer: "IKEA of Sweden", model: "FYRTUR block-out roller blind", deviceJoinName: "IKEA Window Treatment" // raw description 01 0104 0202 01 09 0000 0001 0003 0004 0005 0020 0102 1000 FC7C 02 0019 1000 //IKEA FYRTUR Blinds
+
+		// Yookee yooksmart
+		fingerprint inClusters: "0000,0001,0003,0004,0005,0102", outClusters: "0019", manufacturer: "Yookee", model: "D10110", deviceJoinName: "Yookee Window Treatment"
+        fingerprint inClusters: "0000,0001,0003,0004,0005,0102", outClusters: "0019", manufacturer: "yooksmart", model: "D10110", deviceJoinName: "yooksmart Window Treatment"
```
Comment (MWierzbinskaS) [inline devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Please fix spacing - use 1 tab instead of 4 spaces - it applies to all changes if necessary

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy [hunk 2]
```diff
@@ -122,6 +129,8 @@ def parse(String description) {
 		} else if (reportsBatteryPercentage() && descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && zigbee.convertHexToInt(descMap?.attrId) == BATTERY_PERCENTAGE_REMAINING && descMap.value) {
 			def batteryLevel = zigbee.convertHexToInt(descMap.value)
 			batteryPercentageEventHandler(batteryLevel)
+		} else if (reportBatteryPercentage() && descMap?.clusterInt == CLUSTER_BATTERY_LEVEL && descMap.value) {
+		    batteryPercentageEventHandler(zigbee.convertHexToInt(descMap.value))
```
Comment (MWierzbinskaS) [inline devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: zigbee.POWER_CONFIGURATION_CLUSTER and CLUSTER_BATTERY_LEVEL are the same (0x0001) clusters. I am not sure how Your device sends reports (is it using BATTERY_PERCENTAGE_REMAINING - 0x0021 attribute), but I think it is worth checking to avoid code duplication.

**Review by MWierzbinskaS — CHANGES_REQUESTED** [review]

**Comment by k-morawski** [comment]
Dear @JeremeLau, could you please check @MWierzbinskaS comments? Please let us know if you need our help with making the changes. Many thanks!

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 1 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Ok, I'll fix the spacing.

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 2 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Since I double checked with our embedded engineers, and it turned out a mistake. I'll fix the code duplication by removing the adding codes

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy [hunk 3]
```diff
@@ -132,6 +138,8 @@ def levelEventHandler(currentLevel) {
 	if (lastLevel == "undefined" || currentLevel == lastLevel) { //Ignore invalid reports
 		log.debug "Ignore invalid reports"
 	} else {
+		state.invalidSameLevelEvent = true
```
Comment (KKlimczukS) [inline devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Please, could You explain why did You set this state variable and never use it in the code ?
(I think this one is used in  another DTH only zigbee-window-shade.groovy)

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy [hunk 4]
```diff
@@ -149,6 +157,7 @@ def levelEventHandler(currentLevel) {
 def updateFinalState() {
 	def level = device.currentValue("level")
 	log.debug "updateFinalState: ${level}"
+	level = level < 0 ? 0 : level > 100 ? 100 : level
```
Comment (KKlimczukS) [inline devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Please, could You explain why did You add this line ? Do Your devices report such values that need to be constrained that way?

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy [hunk 5]
```diff
@@ -132,6 +138,8 @@ def levelEventHandler(currentLevel) {
 	if (lastLevel == "undefined" || currentLevel == lastLevel) { //Ignore invalid reports
 		log.debug "Ignore invalid reports"
 	} else {
+		state.invalidSameLevelEvent = true
+		currentLevel = currentLevel < 0 ? 0 : currentLevel > 100 ? 100 : currentLevel
```
Comment (KKlimczukS) [inline devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Please, could You explain why did You add this line ? Do Your devices report such values that need to be constrained that way?

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 3 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Yes, It's a another PR problem, I'll remove the code.

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 4 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Yes, it's useful for some condition in compatible with Alexa and Google.

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 5 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: For 'invalidSameLevelEvent', I'll make this right.
For next line, It's necessary as in some condition compatible with Alexa and Google.

Comment (KKlimczukS) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Are You saying that it's useful when You ask Alexa or Google to set values 0 <  and > 100 or does it happen in other conditions ?

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 4 above)
Comment (KKlimczukS) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: Setting 0 and 100 will not work here. The next condition will not let You send any event in such cases.

**Review by KKlimczukS — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 4 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: I have removed the code

File: devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy (hunk 5 above)
Comment (JeremeLau) [reply devicetypes/smartthings/zigbee-window-shade-battery.src/zigbee-window-shade-battery.groovy]: I have removed the code

**Review by KKlimczukS — APPROVED** [review]
